api-environment/src/main/java/com/synopsys/integration/alert/environment/EnvironmentProcessingResult.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * api-environment
+ *
+ * Copyright (c) 2022 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.alert.environment;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import com.synopsys.integration.util.Stringable;
+
+public class EnvironmentProcessingResult extends Stringable {
+    private final Set<String> variableNames;
+    private final Map<String, String> variableValues;
+
+    public static EnvironmentProcessingResult empty() {
+        return new EnvironmentProcessingResult(Set.of(), Map.of());
+    }
+
+    private EnvironmentProcessingResult(Set<String> variableNames, Map<String, String> variableValues) {
+        this.variableNames = variableNames;
+        this.variableValues = variableValues;
+    }
+
+    public boolean hasValues() {
+        return !variableValues.isEmpty();
+    }
+
+    public Set<String> getVariableNames() {
+        return variableNames;
+    }
+
+    public Optional<String> getVariableValue(String variableName) {
+        return Optional.ofNullable(variableValues.getOrDefault(variableName, null));
```

Comment:
Is this redundant? Wouldn't a map return null if there was not variable?

Reply:
yes.  I was going to return a different default but didn't need to.